### PR TITLE
[docs] Add polish to the date-picker playground

### PR DESCRIPTION
**Pull request by zanivan**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

While checking out the X docs, I noticed that the playground borders seemed too bright against the dark theme, which was a bit off. So, I decided to tweak them a bit and ended up making some other adjustments along the way.

| Current | New |
|--------|--------|
| Screenshot 2024-01-30 at 16 02 19 | Screenshot 2024-02-07 at 11 37 49 | 

👉 https://deploy-preview-11869--material-ui-x.netlify.app/x/react-date-pickers/date-picker/#customization

### CODE REVIEW

**Comment by zanivan** [comment]
@noraleonte I just made some stray tweaks, please let me know if anything needs to be fixed, since this is my first PR for X :)

File: docs/data/date-pickers/date-picker/CustomizationExamplesNoSnap.js [hunk 1]
```diff
@@ -28,13 +28,14 @@ export default function CustomizationExamplesNoSnap() {
...
     <LocalizationProvider dateAdapter={AdapterDayjs}>
-      <Stack spacing={2} sx={{ mb: 2, width: '100%' }}>
+      <Stack spacing={2} sx={{ mb: 2, width: '100%', px: { xs: 2, sm: 0 } }}>
         <BrandingProvider>
           <ToggleButtonGroup
             value={selectedPicker}
             exclusive
             onChange={handleSelectedPickerChange}
             aria-label="date picker components"
+            sx={{}}
```
Comment (noraleonte) [inline docs/data/date-pickers/date-picker/CustomizationExamplesNoSnap.js:38]: Not sure if this is needed 🤔

File: docs/src/modules/components/CustomizationPlayground.tsx [hunk 2]
```diff
@@ -400,7 +405,25 @@ const CustomizationPlayground = function CustomizationPlayground({
...
+              value={selectedCustomizationOption}
+              options={customizationOptions}
+            />
+            {selectedExample && (
+              <StylingRecommendation
+                type={selectedExample.type}
+                message={selectedExample?.comments}
+              />
+            )}
+          </TabsWrapper>{' '}
```
Comment (noraleonte) [inline docs/src/modules/components/CustomizationPlayground.tsx:424]:
```suggestion
          </TabsWrapper>
```

File: typescript
Comment (noraleonte) [inline typescript:1]: Do we need this file? 🤔

**Review by noraleonte — COMMENTED** [review]
Thanks for picking this up, @zanivan 💯 🎉 
Looks great, I just left some very tiny nitpicks 😄

**Comment by zanivan** [comment]
Just a little bump here @noraleonte :)

**Review by noraleonte — APPROVED** [review]
Looks great! Thanks for taking care of this! 💙

**Pull request merged**